src/Riverline/DynamoDB/Connection.php

```diff
@@ -59,15 +62,25 @@ public function __construct($key, $secret, $region, $version = 'latest')
      * @return Connection
      * @throws ConfigurationException
      */
-    static public function CreateFromDynamoDbClient(DynamoDbClient $dynamoDbClient)
+    public static function CreateFromDynamoDbClient(DynamoDbClient $dynamoDbClient)
     {
-        $apiVersion = $dynamoDbClient->getApiVersion();
-        if($apiVersion != '2011-12-05') {
-            throw new ConfigurationException("Api version '{$apiVersion}' is not supported");
+        $apiVersion = $dynamoDbClient->getApi()->getApiVersion();
+
+        if ($apiVersion !== self::SUPPORTED_API_VERSION) {
```

Comment:
Given the default for the constructor is `latest`, I'd suggest binning off this check. I know I suggested to keep it in the first place, but that's before we looked at it last night - there doesn't seem to be any reason why it has to be a specific version.